### PR TITLE
feat: categorytree-cache-implementation

### PR DESCRIPTION
**Pull request by kibo-sushant**

### CODE REVIEW

File: lib/gql/client/index.ts [hunk 1]
```diff
@@ -5,3 +5,10 @@ export function makeGraphQLClient(endpoint?: string) {
     endpoint || `${process.env.NEXT_PUBLIC_URL ? process.env.NEXT_PUBLIC_URL : ''}/api/graphql`
   return new GraphQLClient(endpoint)
 }
+
+export function makeCategoryTreeGraphQLClient(endpoint?: string) {
```
Comment (kibo-kevinwatts) [inline lib/gql/client/index.ts:9]: do we need this? We're just doing a fetch to an api endpoint? we don't need a gql client

**Review by kibo-kevinwatts — CHANGES_REQUESTED** [review]
overall this looks good

**Review by TriptiVermaXP — APPROVED** [review]

File: lib/api/operations/get-category-tree.ts [hunk 2]
```diff
@@ -1,10 +1,24 @@
+import getConfig from 'next/config'
+
 import { fetcher } from '@/lib/api/util'
+import cache from '@/lib/api/util/cache'
 import { getCategoryTreeQuery } from '@/lib/gql/queries'
 
+const { serverRuntimeConfig } = getConfig()
+const key = serverRuntimeConfig.cacheKey
```
[the remark this answers is not in the document]
Comment (kibo-sushant) [reply lib/api/operations/get-category-tree.ts]: Resolved

File: lib/api/operations/get-category-tree.ts [hunk 3]
```diff
@@ -1,10 +1,24 @@
...
-    const response = await fetcher({ query: getCategoryTreeQuery, variables: {} }, null)
-    return response.data.categoriesTree?.items
+    const cachedItems = cache.get(key)
+    if (cachedItems) return cachedItems
+
+    if (!cachedItems) {
+      const response = await fetcher({ query: getCategoryTreeQuery, variables: {} }, null)
+      cache.set(key, response, timeOut)
+
+      return response
```
[the remark this answers is not in the document]
Comment (kibo-sushant) [reply lib/api/operations/get-category-tree.ts]: Resolved

File: lib/gql/client/index.ts (hunk 1 above)
Comment (kibo-sushant) [reply lib/gql/client/index.ts]: Resolved using browser fetch, I don't think vercel-fetch still ready to use in browser
image

**Review by kibo-kevinwatts — APPROVED** [review]

**Pull request merged**